app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,72 @@
+class PostsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :correct_user, only: :destroy
+
+  def index
+    if params[:search].present?
+      Post.reindex
```

Comment:
This should be in model layout, and should run when post creared/updated. Not when we try to search something.